src/c.c

```diff
@@ -90,7 +89,7 @@ SEXP vec_c(SEXP xs,
 
     init_compact_seq(idx_ptr, counter, size, true);
 
-    out = vec_proxy_assign_opts(out, idx, elt, &c_assign_opts);
+    out = vec_proxy_assign_opts(out, idx, elt, vctrs_ownership_owned, &c_assign_opts);
```

Comment:
Needs a comment that we own the `out` proxy because it was taken of a fresh `vec_init()`. Same for bind ops etc.